### PR TITLE
First push of the Power Source Cluster.

### PR DESCRIPTION
**Pull request by JimBuzbee**
First push of the Power Source Cluster. Derived from project-chip XML Note prefix added to BatCommonDesignationEnum keys as Spec called for numeric keys which are illegal under Typescript. Untested.

### CODE REVIEW

File: src/matter/cluster/PowerSourceCluster.ts [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+    name: "Power Source",
+    revision: 1,
+
+    /** @see {@link MatterCoreSpecificationV1_0} § 11.7.4 */
+    features: {
+      /**  Wired power source */
+      wired: BitFlag(0),
+
+      /** Battery power source */
+      Battery: BitFlag(1),
```
Comment (Apollon77) [inline src/matter/cluster/PowerSourceCluster.ts:295]: lowercase please, also below

File: src/matter/cluster/PowerSourceCluster.ts [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+      Rechargeable: BitFlag(2),
+
+      /** Replaceable battery power source (requires Battery feature) */
+      Replaceable: BitFlag(3),
+    },
+
+    /** @see {@link MatterCoreSpecificationV1_0} § 11.7.6.1 */
+    attributes: {
+      /** SHALL indicate the participation of this power source in providing power to the Node. */
+      Status: Attribute(0x0000,TlvEnum<PowerSourceStatusEnum>()),
```
Comment (Apollon77) [inline src/matter/cluster/PowerSourceCluster.ts:307]: nit space after comma please (we have formatter soon) :-)

Comment (Apollon77) [reply src/matter/cluster/PowerSourceCluster.ts:307]: And also here lowercase please

**Review by Apollon77 — CHANGES_REQUESTED** [review]

File: src/matter/cluster/PowerSourceCluster.ts [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+  NickelZinc = 27,
+  SilverZinc = 28,
+  SodiumIon = 29,
+  SodiumSulfur = 30,
+  ZincBromide = 31,
+  ZincCerium = 32,
+}
+
+/** @see {@link MatterCoreSpecificationV1_0} § Table 99 */
+const WiredFaultChangeEvent = <T>() => ({
```
[the remark this answers is not in the document]
Comment (JimBuzbee) [reply src/matter/cluster/PowerSourceCluster.ts:256]: Yeah 

= <T>() => ({

Kind of looks like a cat walked across the keyboard :-)  Was giving me some trouble satisfying the "event:" definition (as did TlvObject), but I was likely overthinking it.  New push on the way - hopefully got it right.

Good catch on the IDs - Fixed.

**Review by Apollon77 — APPROVED** [review]

**Pull request merged**